### ISSUE TITLE
Worked perfectly first after adding (before restart) - after seems 50% chance to work or not.

### ISSUE THREAD

**Issue by AtmaDarkwolf** [manutalcual/winredock#35]
After reboot, WinReDock loads as expected, but after screen display sleeps, seems random if it bothers working or not (Sometimes it does, other times windows shifts all open windows to left monitor.)

I cannot understand why it would work some of the time but not other times. (Also now and then, when I wake monitors, some, but not all, of the windows I had on right monitor would move back to left, but others were in expected spots)

**Comment by manutalcual** [comment]
Hi,

Thanks for noticing this!
I'm going to investigate. Now I have access to a Windows machine and I hope I'll have time to invest on this.

**Comment by AtmaDarkwolf** [comment]
if it helps with research on this, its partly cuz 1 monitor is DP and one is HDMI, and (without WinReDock) all open windows will move to the same relative position on the left monitor(DP, main), partly due to the gpu 'turning off' the port to both monitors (I can hear the pc 'disconnect' then reconnect sound when I wake monitors back up)

Does so even when I have it set to 'go to lock screen' (IE need to relog back in) 

The (so far anyways) programs I find most affected by it is steam (small mode, friends list, and chat windows) some, or all do not move back to main screen (always at least 1 of the 3 steam based windows is stuck on left monitor) and rarely discord will move back over, but it remembers position of windows explorer windows, and most of the time keeps firefox windows where they were (have 2 windows with multiple tabs open, one on each monitor)

Regardless of issues, thank you for this app, it has (even if only partly) solved an issue windows has had since before windows 7 (Which was easily fixed with reg-edits until win10)

**Comment by bfeist** [comment]
Thanks for the great piece of software.
Like these other posts, I'm experiencing the same thing. Hit or miss whether winredock will do its thing.

This app is a life saver. I hope you're able to solve this issue. Thanks again.

**Comment by manutalcual** [comment]
Hi,

Thanks for your kind words man!!

**TL;DR** Some windows can be rescaled, some can't because they handle that themselves. Winredoc doesn't knows who rescales and who don't, so we don't rescale as doing nothing keeps everything viewable and working (even on the wrong screen) and rescaling an already scaled window makes it unmanageable and unviewable (on every screen.)

I've been investigating this DPI problem (and tool windows, read below) and found that it doesn't have a solution (I can found) because of how the DPI is managed.
May be that's why Windows itself isn't implementing this.
The problem is because applications handle DPI changes is several ways: ignoring it (then Windows will rescale the application **content** like if they were images), the application taking care of the DPI change event and changing its size (**but not rescaling** the app drawings) and the app taking the DPI change event and rescaling some or all of its drawings.
Rescaling the drawings means not only to resize and reposition "controls" (buttons, images, lists, etc.), but **also the image** that forms the "control" itself.
Due to this fact, and to the fact that no one else can know if the app rescaled everything needed, Winredoc can't scale the windows because they may become "unviewable" (images that form the "controls" become highly distorted.) (e.g. font too smal to read, minimize, maximize and/or other button icons being huge, but very very HUGE in screen, to the point it takes a 1/4 of screen.)

I've tried everything.

In one moment I thought I have it, but I discovered Visual Studio was rescaling everything twice: one because of Windows DPI change event and another one because Winredoc was either, rescaling the window or sending the rescale event again.

As for (those small mode, friends list, and chat) tool windows... it is the same pain, there are windows that are "visible" (in terms of internal programming) and those that don't. Winredock can't move tool windows that are not "registered" as true windows (there are several way to register windows in a system.)

I guess we have to wait until every app modernizes and handles the rescaling themselves and there is only ONE way to do it.
AND I'm pretty sure Microsoft will then make Windows reposition the windows much better than me.

**Comment by bfeist** [comment]
For what it's worth, I've turned to trying other things and found that DisplayFusion Pro is able to rearrange all windows into a saved profile arrangement. In this app I have to use a bound key combination to reset the view, but it works every time. Maybe you could figure out how they're doing it and implement that? I would much rather have it happen automatically as winredock does it without all of the unneeded extra features (and price) of DisplayFusion.

**Comment by manutalcual** [comment]
Hmm.

I don't quite understand what DF does. It saves one "windows position configuration" and restores it when key combination is pressed?
If so, they don't have to handle DPI at all. They change positions *after* the DPI change. Winredock moves windows *at the moment* of the screen resolution change, most times (but not all) just *before* the app receiving the DPI change.
It is very hard to get this well as there are several ways to get the window position/size depending on several factors (physical screen, extended desktop, virtual desktop, DPI changes, etc.)
I'm just thinking I'm going to put a small delay before changing window positions to see if they get the DPI change before I move them and see if it works better.

I'll continue to investigate a bit, but sincerely, I don't use Ms Windows myself, I use Linux.
And DisplayFusion gets money for their investigation, something I do not.

**Comment by bfeist** [comment]
Yes, it saves all window locations into a "profile" and then restores that when you command it to. Here's a screenshot of one of my window profiles in case you find it useful. 
![image](https://user-images.githubusercontent.com/662952/107981799-cbcb3c80-6f90-11eb-96fa-f7b419367b14.png)
